packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -0,0 +1,314 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as cdk from '@aws-cdk/core';
+import { CfnCanary } from './synthetics.generated';
+
+/**
+ * Properties for a canary
+ */
+export interface CanaryProps {
+  /**
+   * The s3 location that stores the data of the canary runs.
+   *
+   * @default - A new s3 bucket will be created.
+   */
+  readonly artifactBucket?: s3.IBucket;
+
+  /**
+   * Canary execution role.
+   *
+   * This is the role that will be assumed by the canary upon execution.
+   * It controls the permissions that the canary will have. The Role must
+   * be assumable by the 'lambda.amazonaws.com' service principal.
+   *
+   * The default Role automatically has permissions granted for Canary execution.
+   * If you provide a Role, you must add the relevant policies.
+   *
+   * The relevant policies are "s3:PutObject", "s3:GetBucketLocation", "s3:ListAllMyBuckets",
+   * "cloudwatch:PutMetricData", "logs:CreateLogGroup", "logs:CreateLogStream", "logs:PutLogEvents".
+   * @see required permissions: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-synthetics-canary.html
+   *
+   * @default - A unique role will be generated for this canary.
+   * Both supplied and generated roles can always be changed by calling 'addToRolePolicy'.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * How many seconds the canary should run before timing out. Minimum time is 60 seconds and maximum is 900 seconds.
+   *
+   * @default - the minimum between frequency and 900 seconds
+   */
+  readonly timeout?: cdk.Duration;
+
+  /**
+   * The amount of memory that is allocated to your Canary. Must be a multiple of 64 MiB and inside the range 960 MiB to 3008 MiB.
+   *
+   * @default - 960 MiB
+   */
+  readonly memorySize?: cdk.Size;
+
+  /**
+   * How long the canary will be in a 'RUNNING' state. For example, if you set `timeToLive` to be 1 hour and `frequency` to be 10 minutes,
+   * your canary will run at 10 minute intervals for an hour, for a total of 6 times.
+   *
+   * @default - no limit
+   */
+  readonly timeToLive?: cdk.Duration;
+
+  /**
+   * How often the canary runs. For example, if you set `frequency` to 10 minutes, then the canary will run every 10 minutes.
+   *
+   * @default Duration.minutes(5)
+   */
+  readonly frequency?: cdk.Duration;
+
+  /**
+   * Whether or not the canary should start after creation. If set to false, then the canary will not start.
+   *
+   * @default true
+   */
+  readonly enable?: boolean;
+
+  /**
+   * How many days should successful runs be retained
+   *
+   * @default Duration.days(31)
+   */
+  readonly successRetentionPeriod?: cdk.Duration;
+
+  /**
+   * How many days should failed runs be retained
+   *
+   * @default Duration.days(31)
+   */
+  readonly failureRetentionPeriod?: cdk.Duration;
+
+  /**
+   * The name of the canary. This constitutes the physical ID of the canary.
```

Comment:
```suggestion
   * The name of the canary. 
```

Reply:
Add some of the details mentioned in the canary docs regarding the name, it's important to mention that this name shouldn't  include secrets or proprietary information as it a part of the canary ARN, and the ARN is included in outbound calls over the internet.